src/AI-Algorithms-Graph/AIPrim.class.st

```diff
@@ -0,0 +1,58 @@
+Class {
+	#name : #AIPrim,
+	#superclass : #AIGraphAlgorithm,
+	#category : #'AI-Algorithms-Graph'
+}
+
+{ #category : #configuration }
+AIPrim >> edgeClass [ 
+	^ AIWeightedEdge
+]
+
+{ #category : #initialization }
+AIPrim >> initialize [ 
+	super initialize.
+]
+
+{ #category : #accessing }
+AIPrim >> minNode [
+	|lowNode lowKey|
+	lowKey := Float infinity.
+	nodes do: [ :each | 
+			((each distance < lowKey) & (each visited not) ) ifTrue: [ lowNode := each ]
+		 ].
+	^lowNode
+]
+
+{ #category : #configuration }
+AIPrim >> nodeClass [ 
+	^ AIBFSNode 
+]
+
+{ #category : #running }
+AIPrim >> run [ 
+	|firstNode numNodes curNode curEdge treeEdges|
+	nodes do: [ :each | each distance: Float infinity ].
+	
+	treeEdges := OrderedCollection new.
+	firstNode := nodes first.
+	firstNode distance: 0.
+
+	numNodes := nodes size.
+
+	numNodes timesRepeat: [ 
+			curNode := self minNode.
+			edges do: [ :edge |
+					curEdge := edge asTuple.
+					(curEdge first == curNode) ifTrue: [ 
```

Comment:
This is the line causing the trouble, verified using the debugger. even though the nodes are equal, it isn't recognising them as equal nodes.